include/boost/geometry/algorithms/detail/overlay/get_turn_info_helpers.hpp

```diff
@@ -278,34 +404,49 @@ class intersection_info
     // TODO: it's more like is_spike_ip_p
     inline bool is_spike_p() const
     {
+        if (base::m_range_p.size() == 2u)
+        {
+            return false;
+        }
         if (base::sides().pk_wrt_p1() == 0)
         {
+            // p:  pi--------pj--------pk
+            // or: pi----pk==pj
+
             if (! is_ip_j<0>())
             {
                 return false;
             }
 
             int const qk_p1 = base::sides().qk_wrt_p1();
             int const qk_p2 = base::sides().qk_wrt_p2();
-                
+
             if (qk_p1 == -qk_p2)
             {
                 if (qk_p1 == 0)
                 {
-                    return is_spike_of_collinear(base::pi(), base::pj(),
-                                                 base::pk());
+                    // qk is collinear with both p1 and p2,
+                    // verify if pk goes backwards w.r.t. pi/pj
+                    return direction_code(base::rpi(), base::rpj(), base::rpk()) == -1;
                 }
-                        
+
+                // qk is at opposite side of p1/p2, therefore
+                // p1/p2 (collinear) are opposite and form a spike
                 return true;
             }
         }
         
         return false;
     }
 
-    // TODO: it's more like is_spike_ip_q
     inline bool is_spike_q() const
     {
+        if (base::m_range_q.size() == 2u)
```

Comment:
Right, here it is too. Is the reasoning that if `base::m_range_q.size() == 2u` then there is no `base::rqk()`?